enhancements/update/cvo-log-level-api.md

```diff
@@ -0,0 +1,325 @@
+---
+title: cvo-log-level-api
+authors:
+  - "@Davoska"
+reviewers:
+  - "@wking"
+  - "@LalatenduMohanty"
+  - "@petr-muller"
+approvers:
+  - "@wking"
+api-approvers:
+  - "@deads2k"
+  - "@bparees"
+  - "@JoelSpeed"
+creation-date: 2023-10-09
+last-updated: 2024-09-26
+tracking-link:
+  - https://issues.redhat.com/browse/OTA-923
+see-also:
+  - "None"
+replaces:
+  - "None"
+superseded-by:
+  - "None"
+---
+
+# CVO Log Level API
+
+## Summary
+
+This enhancement proposes to create a new OpenShift API to provide a simple
+method of dynamically changing the verbosity level of Cluster Version Operator's
+logs. There will be four log levels available. The lowest level being the
+current default log level being used by the Cluster Version Operator.
+
+## Motivation
+
+There is currently no way to easily change the verbosity of the [Cluster Version Operator (CVO)](https://github.com/openshift/cluster-version-operator)
+logs in a live cluster.
+
+It would be useful to provide functionality for the cluster administrators and
+OpenShift engineers to easily modify the log level to a desired level using an API
+[similarly as can be done for some other operators](https://github.com/openshift/api/blob/5852b58f4b1071fe85d9c49dff2667a9b2a74841/operator/v1/types.go#L67-L74).
+
+### User Stories
+
+* As an OpenShift administrator, I want to increase the log level of the CVO 
+  from the default level to more easily troubleshoot any potential issues regarding
+  the cluster or the CVO.
+* As an OpenShift administrator, I want to decrease the log level of the CVO
+  from a previously set higher log level to save storage space.
+* As an OpenShift engineer, I want to increase the log level of the CVO for
+  a CI run so that I can more easily troubleshoot any potential issues.
+
+### Goals
+
+* Add a user-facing API for controlling the verbosity of the CVO logs.
+
+### Non-Goals
+
+* Change the default logging verbosity of the Cluster Version Operator.
+* Allow users to set a lower than the current default logging level.
+
+## Proposal
+
+This enhancement proposes to create a new `CustomResourceDefinition` (CRD) 
+called `clusterversionoperators.operator.openshift.io`. The CRD will be part 
+of the [`github.com/openshift/api/operator/v1`][github.com/openshift/api/operator/v1]
+package. A `ClusterVersionOperator` resource will be used to configure the 
+CVO. The configuration, as of now, will only contain the knob to modify the CVO 
+log level. A `ClusterVersionOperator` resource will be added to the OCP 
```

Comment:
How will a CVO know what resource to consume? Likely a well-known name (`cluster`)? CVO is a singleton controller so it seems appropriate.

Reply:
The CVO will know the name of the resource. Yes, most likely it will be `cluster`. I have left the exact name up to implementation details outside of scope of this enhancement. I'll add it to the PR to clear any potential confusion :+1:

Reply:
I appreciate the review.

I have tried to improve the text in https://github.com/openshift/enhancements/pull/1492/commits/2664f1e8dabbff2a2cd42c02d6416576816ada18.

Definitely let me know if it's not clear or if it could be improved.